### PR TITLE
Added account grouping logic

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

**Comment by Mingela** [comment]
Going to rewrite test to spam `transferAsset` transaction instead of `createAccount` as well

File: src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java [hunk 1]
```diff
@@ -86,36 +102,84 @@ public BasicTransactionProvider(IrohaAPI irohaAPI,
...
+              .getTransactionsList();
+
+          if (blockTransactions != null) {
+            // committed transfer transactions
+            blockTransactions.forEach(
+                transaction -> {
+                  if (transaction.getPayload().getReducedPayload().getCommandsList().stream()
+                      .anyMatch(Command::hasTransferAsset)) {
+                    pendingAccounts.remove(getTxAccountId(transaction));
+                  }
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java]: I think it must be refactored

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java [hunk 2]
```diff
@@ -15,6 +15,10 @@
 
   private final IrohaAPI irohaAPI;
   private final KeyPair keyPair;
+  private final KeyPair fakeKeyPair = Utils.parseHexKeypair(
+      "0000000000000000000000000000000000000000000000000000000000000000",
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: Why?

Comment (Mingela) [reply src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: It's just a way to fail transaction in Iroha if it fails validation in BRVS

File: src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java [hunk 3]
```diff
@@ -25,19 +31,27 @@
...
 
   private final IrohaAPI irohaAPI;
+  // BRVS account id to query Iroha
   private final String accountId;
+  // BRVS keypair to query Iroha
   private final KeyPair keyPair;
   private final TransactionVerdictStorage transactionVerdictStorage;
-  private final ObservableRxList<Transaction> cache = new ObservableRxList<>();
+  // Local BRVS cache
+  private final Map<String, List<Transaction>> cache = new HashMap<>();
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java]: Access to these collections doesn't look threadsafe

File: src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java (hunk 2 above)
Comment (dolgopolovwork) [reply src/main/java/iroha/validation/transactions/signatory/impl/TransactionSignerImpl.java]: Can we make it static?

File: src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java (hunk 1 above)
Comment (Mingela) [reply src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java]: Created one more method

**Pull request merged**